### ISSUE TITLE
Exclusion options for file-name-casing

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#4050]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.0.0-RC
- __Running TSLint via__: (pick one) *

#### TypeScript code being linted

`index.ts`:

```ts
console.log("Hello world!");
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "file-name-casing": [
            true,
            "pascal-case"
        ],
    }
}
```

#### Actual behavior

It's common to use names like `index.ts` or `main.ts` in Node. I'd like to be able to exclude file names such as these.

#### Expected behavior

No option exists.

**Comment by rafaelss95** [comment]
@JoshuaKGoldberg is it the same as #4071?
